kornia/nerf/nerf_solver.py

```diff
@@ -22,25 +22,24 @@ class NerfSolver:
         device: device for class tensors: Union[str, torch.device]
         dtype: type for all floating point calculations: torch.dtype
     """
+    _nerf_model: Module
+    _opt_nerf: optim.Optimizer
 
-    def __init__(self, device: Device, dtype: torch.dtype) -> None:
+    def __init__(self, device: torch.device, dtype: torch.dtype) -> None:
```

Comment:
why not adopting out Device stuff ?

Reply:
Specifically, this one is because `nerf_model.to(...)` expects `None`, `int` or `torch.device` -- and `kornia.core.Device` is `str or torch.device` so, is incompatible.


```output
kornia/nerf/nerf_solver.py:98: error: Argument "device" to "to" of "Module" has incompatible type
"Union[str, device]"; expected "Union[int, device, None]"  [arg-type]
            self._nerf_model.to(device=self._device, dtype=self._dtype)
```